processor/stream/test_approved_stream_result/testIntegrationResultInvalidMetadata.approved.json

```diff
@@ -3,7 +3,7 @@
     "errors": [
         {
             "document": "{\"metadata\": {\"user\": null}}",
-            "message": "failed to validate metadata: error validating JSON: I[#] S[#] doesn't validate with \"metadata#\"\n  I[#] S[#/required] missing properties: \"service\""
+            "message": "validation error 'metadata' required"
```

Comment:
This error message doesn't seem right? We have metadata, it's just invalid.

Reply:
The validation for structs checks if any of its attributes is set; since only `user` is given and it is null, `metadata` are treated as not set.

Reply:
OK, I see. I think it would be ideal if we could identify this difference and report a different error. Otherwise it could make for some confusing error messages. Not now, and not high priority.

Reply:
Yeah, that's going to be tricky, we would need to either run the validation directly on the decoding (which I avoided to make it easier to switch between different decoders - e.g. simdjson in the future); or we would need to also introduce a state that checks whether or not a struct was set, similar to what I do with `IsSet()` in the `nullable` package. I started out with something like that, but it also gets confusing easily, e.g. for semantic difference of `nil`, `not set`, `zero` etc. and how this should be reflected on the input models.